core/domain/platform_feature_services.py

```diff
@@ -153,8 +153,10 @@ def _get_server_mode():
         mode, 'prod' if in production mode.
     """
     return (
-        platform_parameter_domain.SERVER_MODES.dev
-        if constants.DEV_MODE else platform_parameter_domain.SERVER_MODES.prod)
+        platform_parameter_domain.SERVER_MODES.dev.value
+        if constants.DEV_MODE
+        else platform_parameter_domain.SERVER_MODES.prod.value
+    )
```

Comment:
Should we return the enum here instead of the value and only transform it to value further down the road.

Reply:
Yeah, I think we should only transform it further down the road. Lemme try this out.

Reply:
This seems like a nice idea. @aasiffaizal have you had any success trying to address this comment?

Reply:
Have implemented it partially, fixing the tests that are breaking. Got caught up with the gsoc deadline. Will finish this soon.

Reply:
done